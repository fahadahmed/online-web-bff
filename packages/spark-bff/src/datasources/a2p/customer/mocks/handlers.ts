import { getAdminOverviewMock } from './getAdminOverview';
import { getCustomerOverviewMock } from './getCustomerOverview';
import { getContentProvidersMock } from './getContentProviders';
import { getServiceRequestsMock } from './getServiceRequests';
import { getShortcodesMock } from './getShortcodes';
import { getShortcodeMock } from './getShortcode';
import { getUsageMock } from './getUsage';
import { addContentProviderMock } from './addContentProvider';
import { addServiceRequestMock } from './addServiceRequest';
import { modifyShortcodeMock } from './modifyShortcode';
import { modifyContentProviderMock } from './modifyContentProvider';
import { deleteContentProviderMock } from './deleteContentProvider';

const handlers = [
  addContentProviderMock,
  addServiceRequestMock,
  getAdminOverviewMock,
  getCustomerOverviewMock,
  getContentProvidersMock,
  getServiceRequestsMock,
  getShortcodesMock,
  getShortcodeMock,
  getUsageMock,
  modifyShortcodeMock,
  modifyContentProviderMock,
  deleteContentProviderMock,
];

export default handlers;
