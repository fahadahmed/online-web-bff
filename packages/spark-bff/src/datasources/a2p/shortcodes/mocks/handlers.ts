import { getAdminShortcodesMock } from './getShortcodes';
import { getAdminShortcodesOverviewMock } from './getShortcodesOverview';

const handlers = [getAdminShortcodesMock, getAdminShortcodesOverviewMock];

export default handlers;
