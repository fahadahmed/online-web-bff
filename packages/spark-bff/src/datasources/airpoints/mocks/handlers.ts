import { rest } from 'msw';
import airpointsResponseMock from './airpoints-response.mock.json';
import addAirpointsResponseMock from './add-airpoints-response.mock.json';
import deleteAirpointsResponseMock from './delete-airpoints-response.mock.json';
import airpointsInactiveResponseMock from './airpoints-inactive-response.mock.json';
import addAirpointsErrorResponseMock from './add-airpoints-error-response.mock.json';
import deleteAirpointsErrorResponseMock from './delete-airpoints-error-response.mock.json';

interface AddAirpointsMockBody {
  accountNumber?: string;
  airpointsNumber?: string;
  firstName?: string;
  lastName?: string;
  status?: string;
}

export const airpointsResponseMocks = rest.get<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber === '123456789') {
      /* Simulate error responses */
      return res(ctx.status(200), ctx.json(airpointsResponseMock));
    }
    return res(ctx.status(200), ctx.json(airpointsInactiveResponseMock));
  },
);

export const updateAirpointsResponseMocks = rest.post<AddAirpointsMockBody>(
  `http://testhost/v1/services/airpoints/:accountNumber`,
  async (
    { params: { accountNumber }, body: { airpointsNumber } },
    res,
    ctx,
  ) => {
    /* add airpoints */
    if (airpointsNumber) {
      if (airpointsNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(400), ctx.json(addAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(addAirpointsResponseMock));
    }
    /* delete airpoints */
    if (accountNumber) {
      if (accountNumber === '911') {
        /* Simulate error responses */
        return res(ctx.status(200), ctx.json(deleteAirpointsErrorResponseMock));
      }
      return res(ctx.status(200), ctx.json(deleteAirpointsResponseMock));
    }
    return null;
  },
);

const handlers = [airpointsResponseMocks, updateAirpointsResponseMocks];

export default handlers;
