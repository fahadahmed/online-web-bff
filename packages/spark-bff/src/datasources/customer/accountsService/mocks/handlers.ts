import { rest } from 'msw';
import accountSummaryMock from './account-summary.mock.json';
import emptyAccountListMock from './empty-account-list.mock.json';
import singleAccountListMock from './single-account-list.mock.json';
import accountListMock from './account-list.mock.json';
import guestAccountListMock from './guest-account-list.mock.json';
import unauthorizedAccountListMock from './unauthorized-account-list.mock.json';

const baseUrl = 'http://testhost/v1/customer/account';

export const accountSummaryHandler = rest.get<unknown, { accountId: string }>(
  `${baseUrl}/:accountId/summary`,
  async (req, res, ctx) => {
    const { accountId } = req.params;
    if (accountId.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountSummaryMock));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

export const accountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(accountListMock));
  },
);

export const singleAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(singleAccountListMock));
  },
);

export const emptyAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(200), ctx.json(emptyAccountListMock));
  },
);

export const unauthorizedAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(401), ctx.json(unauthorizedAccountListMock));
  },
);

export const guestAccountListHandler = rest.get(
  `${baseUrl}/me/list`,
  async (_, res, ctx) => {
    return res(ctx.status(403), ctx.json(guestAccountListMock));
  },
);

const handlers = [accountSummaryHandler, accountListHandler];

export default handlers;
