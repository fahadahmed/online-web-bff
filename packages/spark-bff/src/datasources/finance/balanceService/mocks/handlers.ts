import { rest } from 'msw';
import accountBalanceDetails from './account-balance.mock.json';

const baseUrl = 'http://testhost/v1/finance/balance/';

export const accountBalanceMock = rest.get<unknown, { accountNumber: string }>(
  `${baseUrl}account/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    if (accountNumber.match(/^\d+$/) !== null) {
      return res(ctx.status(200), ctx.json(accountBalanceDetails));
    }
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 713012977, then it returns failure response, "Bad Request" */
    return res(ctx.status(400), ctx.json({}));
  },
);

const handlers = [accountBalanceMock];

export default handlers;
