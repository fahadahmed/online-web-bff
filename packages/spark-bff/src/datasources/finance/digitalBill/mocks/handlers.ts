import { rest } from 'msw';
import accountBillingSummary from './account-billing-summary.mock.json';
import accountTransactions from './account-transactions.mock.json';

const baseUrl = 'http://testhost/v2/finance';

export const accountBillingSummaryMock = rest.get(
  `${baseUrl}/:accountNumber/bills/summary`,
  async (_, res, ctx) => {
    return res(ctx.json(accountBillingSummary));
  },
);

export const accountTransactionsMock = rest.get(
  `${baseUrl}/:accountNumber/bills/transactions`,
  async (_, res, ctx) => {
    return res(ctx.json(accountTransactions));
  },
);

const handlers = [accountBillingSummaryMock, accountTransactionsMock];

export default handlers;
