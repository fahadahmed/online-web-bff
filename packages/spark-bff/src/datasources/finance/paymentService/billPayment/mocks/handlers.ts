import { rest } from 'msw';
import poliPayBillSuccess from './poliPayBill-success.mock.json';
import poliPayBillValidationFailure from './poliPayBill-failure.mock.json';
import creditCardPayBillSuccess from './creditCard-payBill-success.mock.json';
import creditCardPayBillValidationFailure from './creditCard-payBill-failure.mock.json';
import deleteAutoPayBillSuccess from './delete-autoPayBill-success.mock.json';
import deleteAutoPayBillFailure from './delete-autoPayBill-failure.mock.json';

const baseUrl = 'http://testhost/v1/finance/payment';

export const addPoliPayBillMock = rest.post<unknown, { accountNumber: string }>(
  `${baseUrl}/polipay/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if ((accountNumber as string).match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(poliPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(poliPayBillSuccess));
  },
);

export const addCreditCardPayBillMock = rest.post<
  unknown,
  { accountNumber: string }
>(
  `${baseUrl}/creditcard/bill/:accountNumber`,
  async ({ params: { accountNumber } }, res, ctx) => {
    /* If the accountNumber does not match the given pattern, [0-9]+
    for example, 789046383,then it returns failure response, "Request validation failure."  */
    if (accountNumber.match(/^\d+$/) === null) {
      return res(ctx.status(400), ctx.json(creditCardPayBillValidationFailure));
    }
    return res(ctx.status(200), ctx.json(creditCardPayBillSuccess));
  },
);

export const deleteAutoPayBillMock = rest.delete<
  unknown,
  { accountNumber: string; paymentMethodId: string }
>(
  `${baseUrl}/bill/:accountNumber/paymentsetup/:paymentMethodId`,
  async ({ params: { accountNumber, paymentMethodId } }, res, ctx) => {
    /** If accountNumber does not match the given pattern,
     for example, 746738998.
     Or If paymentMethodId does not match the given pattern,
     for example,"1-S0123-23"
     it returns failure response, "Request Validation Failure".
     */
    if (
      accountNumber.match(/^\d+$/) === null ||
      paymentMethodId.match(/^[\dA-Z][\dA-Z-]+[\dA-Z]$/) === null
    ) {
      return res(ctx.status(400), ctx.json(deleteAutoPayBillFailure));
    }
    return res(ctx.status(200), ctx.json(deleteAutoPayBillSuccess));
  },
);

const handlers = [
  addCreditCardPayBillMock,
  addPoliPayBillMock,
  deleteAutoPayBillMock,
];

export default handlers;
