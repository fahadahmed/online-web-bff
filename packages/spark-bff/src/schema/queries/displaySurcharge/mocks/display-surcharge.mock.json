{
  "PAY_BILL": {
    "operationType": "PAY_BILL",
    "masterCard": "3%",
    "visa": "3%",
    "amex": "2%",
    "dinersClub": "2%",
    "displayText": [
      "American Express and Diners Club incur a 2% surcharge.",
      "Visa/MasterCard, Prezzy Card, international credit, debit or charge cards incur a 3% surcharge."
    ]
  },
  "SETUP_MONTHLY_PAYMENT": {
    "operationType": "SETUP_MONTHLY_PAYMENT",
    "masterCard": "3%",
    "visa": "3%",
    "amex": "2%",
    "dinersClub": "2%",
    "displayText": [
      "Automated payments using a credit/debit card incurs a 1.5% surcharge (plus GST, if any). Surcharge applies."
    ]
  }
}
