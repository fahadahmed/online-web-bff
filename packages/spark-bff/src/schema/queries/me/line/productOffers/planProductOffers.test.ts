import { planProductOffersHandler } from 'datasources/product/offers/mocks/handlers';
import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';

test('returns all the product offer details data with no errors', async () => {
  server.use(planProductOffersHandler);

  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query planProductOffersQuery(
        $lineNumber: String!
        $packInstanceId: String
      ) {
        me {
          line(lineNumber: $lineNumber) {
            planProductOffers(packInstanceId: $packInstanceId) {
              offerDetails {
                id
                name
                description
                price {
                  frequency {
                    period
                    value
                  }
                  basePrice
                }
                entitlements {
                  name
                  shortDescription
                  longDescription
                  iconUrl
                  sortOrder
                }
                promotions {
                  name
                  productType
                  shortDescription
                  longDescription
                  iconUrl
                  sortOrder
                }
              }
              notifications {
                recommendationId
                text
                title
                type
              }
            }
          }
        }
      }
    `,
    variables: {
      lineNumber: '0276404520',
      packInstanceId: 'legacy050114',
    },
  });

  expect(res.errors).toBeUndefined();
  expect(res.data).toMatchSnapshot();
});
