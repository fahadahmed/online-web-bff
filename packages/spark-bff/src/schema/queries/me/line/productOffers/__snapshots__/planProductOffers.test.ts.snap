// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns all the product offer details data with no errors 1`] = `
Object {
  "me": Object {
    "line": Object {
      "planProductOffers": Object {
        "notifications": Array [
          Object {
            "recommendationId": "vas_downgrade_reactivate",
            "text": "All your devices will be become unprotected once you change to McAfee Standard. You'll need to choose which devices to protect again.",
            "title": null,
            "type": "WARN",
          },
        ],
        "offerDetails": Array [
          Object {
            "description": "Ideal for online gaming, HD video streaming, file downloads and more",
            "entitlements": Array [
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/prepaid_small.png",
                "longDescription": "Lorem ipsum dolor sit amet, cosectetur adipiscing eli suspendisse. Lorem dolor sit ame.",
                "name": "40GB",
                "shortDescription": "40GB Max Speed Data",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": "Excludes MMS (e.g. picture messaging) and non-data video calling.",
                "name": "300",
                "shortDescription": "300 Rollover mins (NZ only)",
                "sortOrder": 2,
              },
            ],
            "id": "32b96d99-2a34-4fc2-a0a5-111c4d36579e",
            "name": "3GB Rollover Plan",
            "price": Object {
              "basePrice": 39.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
            },
            "promotions": Array [
              Object {
                "iconUrl": null,
                "longDescription": null,
                "name": "Marketing",
                "productType": "MARKETING",
                "shortDescription": "Marketing tag",
                "sortOrder": null,
              },
              Object {
                "iconUrl": null,
                "longDescription": null,
                "name": "Save",
                "productType": "DEAL",
                "shortDescription": "Save up to 20% on multiple plans",
                "sortOrder": null,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/spotify_small.png",
                "longDescription": "Usually $14.99/month",
                "name": "Spotify",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Half Price Spotify Premium",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/plus_small.png",
                "longDescription": "Hotspot counts towards your total data usage on your Mobile plan. One hotspot extra can be added per line per month.",
                "name": "Hotspot",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Hotspot your data for $10/month",
                "sortOrder": 2,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": null,
                "name": "contracts",
                "productType": "CHARACTERISTIC",
                "shortDescription": "No contracts, 30 days notice",
                "sortOrder": 3,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": null,
                "name": "VoiceMail",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Spark Voicemail App",
                "sortOrder": 4,
              },
            ],
          },
          Object {
            "description": "Ideal for online gaming, HD video streaming, file downloads and more",
            "entitlements": Array [
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/prepaid_small.png",
                "longDescription": "Lorem ipsum dolor sit amet, cosectetur adipiscing eli suspendisse. Lorem dolor sit ame.",
                "name": "12GB",
                "shortDescription": "12GB Max Speed Data",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": "Excludes MMS (e.g. picture messaging) and non-data video calling.",
                "name": "Unlimited",
                "shortDescription": "Unlimited AU/NZ Mins & SMS",
                "sortOrder": 2,
              },
            ],
            "id": "1f0b37dc-0987-4aed-b2aa-a8f22c277e46",
            "name": "Endless Mobile Plan",
            "price": Object {
              "basePrice": 59.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
            },
            "promotions": Array [
              Object {
                "iconUrl": null,
                "longDescription": null,
                "name": "Save",
                "productType": "DEAL",
                "shortDescription": "Save up to 20% on multiple plans",
                "sortOrder": null,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/spotify_small.png",
                "longDescription": "Usually $14.99/month",
                "name": "Spotify",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Half Price Spotify Premium",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/plus_small.png",
                "longDescription": "Hotspot counts towards your total data usage on your Mobile plan. One hotspot extra can be added per line per month.",
                "name": "Hotspot",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Hotspot your data for $10/month",
                "sortOrder": 2,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": null,
                "name": "contracts",
                "productType": "CHARACTERISTIC",
                "shortDescription": "No contracts, 30 days notice",
                "sortOrder": 3,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": null,
                "name": "VoiceMail",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Spark Voicemail App",
                "sortOrder": 4,
              },
            ],
          },
          Object {
            "description": "Ideal for online gaming, HD video streaming, file downloads and more",
            "entitlements": Array [
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/prepaid_small.png",
                "longDescription": "Lorem ipsum dolor sit amet, cosectetur adipiscing eli suspendisse. Lorem dolor sit ame.",
                "name": "40GB",
                "shortDescription": "40GB Rollover Data",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": "Excludes MMS (e.g. picture messaging) and non-data video calling.",
                "name": "Unlimited",
                "shortDescription": "Unlimited AU/NZ Mins & SMS",
                "sortOrder": 2,
              },
            ],
            "id": "f8f6d39c-32ca-4f85-af08-360bae815a76",
            "name": "Endless Mobile Plan",
            "price": Object {
              "basePrice": 79.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
            },
            "promotions": Array [
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/spotify_small.png",
                "longDescription": "Usually $14.99/month",
                "name": "Spotify",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Half Price Spotify Premium",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/plus_small.png",
                "longDescription": "Hotspot counts towards your total data usage on your Mobile plan. One hotspot extra can be added per line per month.",
                "name": "Hotspot",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Hotspot your data for $10/month",
                "sortOrder": 2,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": null,
                "name": "contracts",
                "productType": "CHARACTERISTIC",
                "shortDescription": "No contracts, 30 days notice",
                "sortOrder": 3,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": null,
                "name": "VoiceMail",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Spark Voicemail App",
                "sortOrder": 4,
              },
            ],
          },
          Object {
            "description": "Ideal for online gaming, HD video streaming, file downloads and more",
            "entitlements": Array [
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/prepaid_small.png",
                "longDescription": "Lorem ipsum dolor sit amet, cosectetur adipiscing eli suspendisse. Lorem dolor sit ame.",
                "name": "40GB",
                "shortDescription": "40GB Rollover Data",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": "Excludes MMS (e.g. picture messaging) and non-data video calling.",
                "name": "Unlimited",
                "shortDescription": "Unlimited AU/NZ Mins & SMS",
                "sortOrder": 2,
              },
            ],
            "id": "f8f6d39c-32ca-4f85-af08-360bae815a86",
            "name": "Prepaid Value Pack",
            "price": Object {
              "basePrice": 79.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
            },
            "promotions": Array [
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/spotify_small.png",
                "longDescription": "Usually $14.99/month",
                "name": "Spotify",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Half Price Spotify Premium",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/plus_small.png",
                "longDescription": "Hotspot counts towards your total data usage on your Mobile plan. One hotspot extra can be added per line per month.",
                "name": "Hotspot",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Hotspot your data for $10/month",
                "sortOrder": 2,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": null,
                "name": "contracts",
                "productType": "CHARACTERISTIC",
                "shortDescription": "No contracts, 30 days notice",
                "sortOrder": 3,
              },
            ],
          },
          Object {
            "description": "Ideal for online gaming, HD video streaming, file downloads and more",
            "entitlements": Array [
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/prepaid_small.png",
                "longDescription": "Lorem ipsum dolor sit amet, cosectetur adipiscing eli suspendisse. Lorem dolor sit ame.",
                "name": "40GB",
                "shortDescription": "40GB Rollover Data",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": "Excludes MMS (e.g. picture messaging) and non-data video calling.",
                "name": "Unlimited",
                "shortDescription": "Unlimited AU/NZ Mins & SMS",
                "sortOrder": 2,
              },
            ],
            "id": "f8f6d39c-32ca-4f85-af08-360bae815a96",
            "name": "Data Lovers Pack",
            "price": Object {
              "basePrice": 79.99,
              "frequency": Object {
                "period": "Month",
                "value": 1,
              },
            },
            "promotions": Array [
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/spotify_small.png",
                "longDescription": "Usually $14.99/month",
                "name": "Spotify",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Half Price Spotify Premium",
                "sortOrder": 1,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/plus_small.png",
                "longDescription": "Hotspot counts towards your total data usage on your Mobile plan. One hotspot extra can be added per line per month.",
                "name": "Hotspot",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Hotspot your data for $10/month",
                "sortOrder": 2,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": null,
                "name": "contracts",
                "productType": "CHARACTERISTIC",
                "shortDescription": "No contracts, 30 days notice",
                "sortOrder": 3,
              },
              Object {
                "iconUrl": "/online/myspark/__dev__/logos/changePlan/check_small.png",
                "longDescription": null,
                "name": "VoiceMail",
                "productType": "CHARACTERISTIC",
                "shortDescription": "Spark Voicemail App",
                "sortOrder": 4,
              },
            ],
          },
        ],
      },
    },
  },
}
`;
